azurerm/config.go

```diff
@@ -627,6 +632,15 @@ func (c *ArmClient) registerEventHubClients(endpoint, subscriptionId string, aut
 	c.eventHubNamespacesClient = ehnc
 }
 
+func (c *ArmClient) registerHDInsightClustersClients(endpoint, subscriptionId string, auth autorest.Authorizer, sender autorest.Sender) {
+	hdInsightClustersClient := hdinsight.NewClustersClientWithBaseURI(endpoint, subscriptionId)
+	setUserAgent(&hdInsightClustersClient.Client)
+	hdInsightClustersClient.Authorizer = auth
+	hdInsightClustersClient.Sender = sender
+	hdInsightClustersClient.SkipResourceProviderRegistration = c.skipProviderRegistration
```

Comment:
_minor_ could we switch over to using the new registration format? `c.configureClient(&hdInsightClustersClient.Client, auth)` (we're doing this gradually for all new resources)